irods/connection.py

```diff
@@ -498,10 +500,12 @@ def _login_pam(self):
             if getattr(self,'DISALLOWING_PAM_PLAINTEXT',True):
                 raise PlainTextPAMPasswordError
 
-        # In general authentication API, a ';' and '=' in the password would be misinterpreted due to those
-        # characters' special meaning in the context string parameter.
-        use_dedicated_pam_api = len(ctx) >= MAX_NAME_LEN or \
-                                {';','='}.intersection(set(new_pam_password))
+        # Normally, we use the AUTH_PLUG_REQ_AN api (generalized to handle both PAM and GSI, as evidenced in the gsi_client_auth_request() method.)
```

Comment:
We don’t need to support GSI anymore if that makes this easier/simpler to refactor/test.

Reply:
GSI authentication was supported through version 4.2.8 of iRODS. With which versions of iRODS will PRC 3.0.0 be compatible?

Reply:
Definitely back to 4.2.8...

So resolving...

Reply:
Ah, I got bad information there. 4.2.6 was the last version of iRODS which supported GSI.

Reply:
Released June 2019.   I think we can purge the kerberos and gsi mentions everywhere in 2025.